### PR TITLE
Prometheus: Add hints to query builder

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
This PR adds hinting to Query builder. We are reusing hinting used in Query field. We will have a **UX ideation session** for improving the UX, but in the meantime I thought that we could add it and implement UI changes in separate PR. 

https://user-images.githubusercontent.com/30407135/153586190-e8aec4fd-fc30-4adc-a146-7ccf994f6469.mov

I have also changed a copy a little so it is more reusable. @achatterjee-grafana could you please have a look. 👀

**Which issue(s) this PR fixes**:
Part of https://github.com/grafana/grafana/issues/44253

**To test:**

1. Enable `promQueryBuilder` feature toggle in `custom.ini`
2. In metric dropdown choose a metric:
  - that includes `_bucket` in name and `hint: add histogram_qunatile()` should be visible. When you click, histogram quantile aggregation operations are added.
  -  that includes `_sum` or `_count` and  `hint: add rate() should be visible.  When you click, histogram quantile aggregation operations are added.
  - that returns a lot of series (21+). Run query with selected metric. `hint: add sum()` should be visible.

### CODE REVIEW

**Review by torkelo — APPROVED** [review]
Looks great! 

Think it makes sense to move to its own component that is all. Would be nice if the buttons are centered, a bit tricky as the padding of the OperationsRow is not uniform to make it so that when operations button margin (needed in case the wrap) don't cause a double button margin. 

So to make the hint buttons centered you might need flex box with align-items: center and marginBottom: theme.spacing(1)

File: public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilder.tsx [hunk 1]
```diff
@@ -8,18 +8,53 @@ import { PrometheusDatasource } from '../../datasource';
...
+import { Button, Tooltip, useStyles2 } from '@grafana/ui';
+import { css } from '@emotion/css';
 
 export interface Props {
   query: PromVisualQuery;
   datasource: PrometheusDatasource;
   onChange: (update: PromVisualQuery) => void;
   onRunQuery: () => void;
   nested?: boolean;
+  series?: DataFrame[];
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilder.tsx]: Updated

File: public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilder.tsx [hunk 2]
```diff
@@ -106,9 +141,18 @@ export const PromQueryBuilder = React.memo<Props>(({ datasource, query, onChange
         {query.binaryQueries && query.binaryQueries.length > 0 && (
           <NestedQueryList query={query} datasource={datasource} onChange={onChange} onRunQuery={onRunQuery} />
         )}
+        {hints && hints?.length > 0 && <div>{hints}</div>}
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilder.tsx]: Yeah, good call. I've moved query hints to newly created PromQueryBuilderHints component.

**Comment by ivanahuckova** [comment]
I've updated styling do the hint buttons are centered with the button. Thanks for the tip!
image

**Comment by torkelo** [comment]
Nice!

**Pull request merged**